nncf/common/pruning/utils.py

```diff
@@ -636,4 +638,43 @@ def identity_mask_propagation(node: NNCFNode, graph: NNCFGraph) -> None:
         # In case for disconnected NNCFGraph
         input_masks = [None]
     assert len(input_masks) == 1
-    node.data['output_mask'] = input_masks[0]
+    input_mask = input_masks[0]
+
+    if isinstance(input_mask, dict):
+        input_mask = find_input_mask_for_node(input_mask, node)
+    node.data['output_mask'] = input_mask
+
+
+def match_multiple_output_masks(output_masks: List[SymbolicMask], output_edges: List[NNCFGraphEdge],
+                                chunk_axis: int) -> Dict['str', SymbolicMask]:
+    """
+    Match multiple input mask to each next nodes.
+
+    :param output_masks: Given output masks.
+    :return: Matched output mask for each next node.
+    """
+    output_shapes = [edge.tensor_shape[chunk_axis] for edge in output_edges]
+    next_nodes = [edge.to_node for edge in output_edges]
+    result_masks = {node.node_name: None for node in next_nodes}
+
+    if len(set(output_shapes)) == 1:
+        for i, split_mask in enumerate(output_masks):
+            result_masks[next_nodes[i].node_name] = split_mask
+    else:
+        for split_mask in output_masks:
+            idx = output_shapes.index(split_mask.shape[0])
+            result_masks[next_nodes[idx].node_name] = split_mask
+            # update already matched
+            output_shapes[idx]=None
+
+    return result_masks
+
+def find_input_mask_for_node(input_mask: Dict['str', SymbolicMask], node: NNCFNode) -> SymbolicMask:
+    """
+    Find exact input mask for node among multiple input masks.
+
+    :param input_mask: Given input masks.
+    :return: Exact input mask for node.
+    """
+    node_name = node.node_name
+    return input_mask[node_name]
```

Comment:
I suggest to use `[]` operator instead of this function

Reply:
Fix.

Reply:
This function could be removed